parsec/core/fs/userfs/userfs.py

```diff
@@ -632,10 +632,43 @@ async def _workspace_minimal_sync(self, workspace_entry: WorkspaceEntry) -> None
             FSBackendOfflineError
         """
         workspace = self.get_workspace(workspace_entry.id)
-        await workspace.minimal_sync(workspace_entry.id)
+
+        # This is a noop if the local workspace manifest is not a placeholder.
+        # If it is a placeholder this means:
+        # 1) either our device has just created the workspace in local
+        # 2) or our device has just processed a sharing message (hence our user
+        #   manifest contains a new workspace entry)
+        # 2') or our device is the creator of the workspace and has previously
+        #     succeeded it minimal sync, but has crashed before realizing so :/
+        # 3) or our device has fetched a new version of the user manifest with a new
+        #   workspace entry in it (i.e. another device of our user has done 1) or 2) )
+        #
+        # Of course only case 1) actually need to do (and will succeed to do) a
+        # minimal sync. In the other cases we will realize the workspace manifest
+        # is already uploaded, start using it instead and return without any remote change.
+        #
+        # However, unlike cases 1) and 2') where only our device knows about the
+        # realm (it is it creator and won't sync/share it until it has done a
+        # minimal sync on it) in cases 2) and 3) we can lose the realm access
+        # (our not having the write access). In such case we consider the minimal
+        # sync has succeeded anyway given our goal is to ensure the workspace
+        # is ready to be used by other devices, and losing (or having a limited)
+        # access is a proof of that.
+        #
+        # Last but not least, in case of bug (or malicious behavior), we could
+        # think of a workspace entry being shared with other users without prior
+        # minimal sync (or realm creation without initial workspace manifest sync).
+        # In such case we fallback into case 1) (if the realm hasn't been created,
+        # the first user to do a minimal sync will create it and become OWNER
+        # of it) or will end up with an access error considered as a minimal
+        # sync success (the famous "task failed successfully").
+        try:
+            await workspace.minimal_sync(workspace_entry.id)
+        except FSWorkspaceNoAccess:
+            pass
```

Comment:
Another approach is to try to download the corresponding remote manifest in order to know whether it is necessary to perform a minimal sync. If this remote manifest is available, it's possible to call `await workspace.transactions.synchronization_step(entry_id, remote_manifest, final=True)` in order to acknowledge its existence although it might not be necessary.